api/src/main/java/org/ccci/idm/user/DefaultUserManager.java

```diff
@@ -380,6 +380,19 @@ public void removeFromGroup(final User user, final Group group) throws DaoExcept
         this.userDao.removeFromGroup(user, group);
     }
 
+    /**
+     * Returns all available groups
+     *
+     * Note that this method is not particular to a user, but is temporarily made available here until a
+     * more suitable framework becomes available for providing group dao.
+     *
+     * @return list of all available groups
+     */
+    @Override
+    public List<Group> getAllGroups() throws DaoException {
```

Comment:
thoughts on including an optional `baseDn` parameter? it could be used to make a call like: `userManager.getAllGroups(googleRootDn)` vs `userManager.getAllGroups(adminGroupsDn)`